c7n/resources/apigw.py

```diff
@@ -284,6 +290,28 @@ def augment(self, resources):
             results.append(r)
         return results
 
+    def get_resources(self, ids, cache=True):
+        deployment_ids = []
+        client = utils.local_session(
+            self.manager.session_factory).client('apigateway')
+        for id in ids:
+            # if we get stage arn, we pick rest_api_id
+            if 'arn:aws:apigateway' in id:
```

Comment:
This might be smoother as `if id.startswith('arn:')` to work outside the standard AWS partition (govcloud, china...)

Reply:
Updated code to use standard AWS partition